### PR TITLE
service/dap: move build error to output event

### PR DESCRIPTION
**Pull request by suzmue**
Since the build error may be long, including it in the error response may not be a good way to get the info to the user. This change moves that information to an output event. It also includes the build command that produced the error.

Updates golang/vscode-go#1425

### CODE REVIEW

**Review by aarzilli — APPROVED** [review]
LGTM

File: service/dap/server.go [hunk 1]
```diff
@@ -723,17 +723,24 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
...
 		switch mode {
 		case "debug":
-			out, err = gobuild.GoBuildCombinedOutput(debugbinary, []string{program}, buildFlags)
+			cmd, out, err = gobuild.GoBuildCombinedOutput(debugbinary, []string{program}, buildFlags)
 		case "test":
-			out, err = gobuild.GoTestBuildCombinedOutput(debugbinary, []string{program}, buildFlags)
+			cmd, out, err = gobuild.GoTestBuildCombinedOutput(debugbinary, []string{program}, buildFlags)
 		}
 		if err != nil {
+			s.send(&dap.OutputEvent{
```
Comment (polinasok) [inline service/dap/server.go:735]: Does this always show up? I vaguely remember we had some issue with the early messages in the session not showing up in the Debug Console.

Comment (suzmue) [reply service/dap/server.go:735]: I think once the initialize request is sent the rest of the messages get logged.

**Review by polinasok — APPROVED** [review]
LGTM

**Pull request merged**